src/overlays/ovl_i2/fox_me.c

```diff
@@ -1,97 +1,2527 @@
+        func_8007BFFC(actor->obj.pos.x - actor->vel.x, actor->obj.pos.y + 30.0f, actor->obj.pos.z - actor->vel.z, 0.0f,
+                      0.0f, 0.0f, 4.0f, 10);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i2/fox_me/func_i2_8018ED9C.s")
+void func_i2_80187C68(Actor* actor, f32 x, f32 y, f32 z, f32 arg4, f32 xRot, f32 yRot, s32 arg7, s32 arg8, s32 id) {
+    Actor_Initialize(actor);
+    actor->obj.status = 1;
+    actor->obj.id = (u16) id;
+
+    actor->obj.pos.x = x;
+    actor->obj.pos.y = y;
+    actor->obj.pos.z = z;
+
+    actor->obj.rot.x = xRot;
+    actor->obj.rot.y = yRot;
+
+    actor->timer_0BC = arg7;
+    actor->timer_0BE = 20;
+    actor->unk_0B4 = arg8;
+    actor->fwork[5] = arg4;
+    Object_SetInfo(&actor->info, actor->obj.id);
+}
+
+void func_i2_80187D08(f32 x, f32 y, f32 z, f32 arg3, f32 xRot, f32 yRot, s32 arg6, s32 arg7, s32 id) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(gActors); i++) {
+        if (gActors[i].obj.status == 0) {
+            func_i2_80187C68(&gActors[i], x, y, z, arg3, xRot, yRot, arg6, arg7, id);
+            return;
+        }
+    }
+}
+
+void func_i2_80187D98(Effect* effect, f32 x, f32 y, f32 z, f32 arg4, bool arg5) {
+    Effect_Initialize(effect);
+    effect->obj.status = 1;
+    effect->obj.id = OBJ_EFFECT_369;
+
+    effect->obj.pos.x = x;
+    effect->obj.pos.y = y;
+    effect->obj.pos.z = z;
+
+    effect->unk_60.z = arg4;
+    effect->unk_48 = arg5;
+    effect->obj.rot.x = 45.0f;
+
+    if (arg5) {
+        effect->obj.rot.x = -45.0f;
+    }
+
+    Object_SetInfo(&effect->info, effect->obj.id);
+}
+
+void func_i2_80187E38(f32 x, f32 y, f32 z, f32 arg3) {
+    s32 i;
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187D98(&gEffects[i], x, y, z, arg3, 0);
+            Audio_PlaySfx(0x31037057U, &gEffects[i].sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            break;
+        }
+    }
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187D98(&gEffects[i], x, y, z, arg3, 1);
+            break;
+        }
+    }
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187D98(&gEffects[i], x, y, z, arg3 + 90.0f, 0);
+            break;
+        }
+    }
+
+    for (i = ARRAY_COUNT(gEffects) - 1; i >= 0; i--) {
+        if (gEffects[i].obj.status == 0) {
+            func_i2_80187D98(&gEffects[i], x, y, z, arg3 + 90.0f, 1);
+            break;
+        }
+    }
+}
+
+void func_i2_80187FF8(Effect* effect, f32 x, f32 y, f32 z) {
+    Effect_Initialize(effect);
+    effect->obj.status = 1;
+    effect->obj.id = OBJ_EFFECT_371;
+
+    effect->obj.pos.x = x;
+    effect->obj.pos.y = y;
+    effect->obj.pos.z = z;
+
+    effect->unk_44 = 0x80;
```

Comment:
```suggestion
    effect->unk_44 = 128;
```
this is an alpha value